Debug build fails on Windows

>  = note: LINK : fatal error LNK1181: cannot open input file 'mimalloc-static.lib'

But the file exists in `target\debug\build\libmimalloc-sys-e3452f81d9314272\out\build\RelWithDebInfo\mimalloc-static.lib`.
It's using the wrong profile: I'm building in debug, but it's using `RelWithDebInfo`.

According to WindowsBunny:
> The build script simplifies down the four cmake targets to a boolean is_debug https://github.com/purpleprotocol/mimalloc_rust/blob/master/libmimalloc-sys/build.rs#L83-L89
> So later when it finds the path to where the resulting library is, it makes invalid assumptions https://github.com/purpleprotocol/mimalloc_rust/blob/master/libmimalloc-sys/build.rs#L107-L119
> I'm actually not sure why it even pushes out_dir onto dst.
> All the examples of using the cmake crate show passing dst directly to cargo without any modification.
> It shouldn't get out_dir based on is_debug.
> Instead it should just check `get_cmake_build_type()` directly and do it based on that (which is robust in case in the future rust ever gains the option to enable linking against the debug CRT and the cmake crate gains support for that).

---

Related: https://github.com/purpleprotocol/mimalloc_rust/issues/10

--- a/libmimalloc-sys/build.rs
+++ b/libmimalloc-sys/build.rs
@@ -76,11 +76,11 @@ fn main() {
     // This set mi_option_verbose and mi_option_show_errors options to false.
     cfg = cfg.define("mi_defines", "MI_DEBUG=0");
 
-    let is_debug = match get_cmake_build_type() {
-        Ok(CMakeBuildType::Debug) => true,
-        Ok(CMakeBuildType::Release) => false,
-        Ok(CMakeBuildType::RelWithDebInfo) => false,
-        Ok(CMakeBuildType::MinSizeRel) => false,
+    let (is_debug, win_folder) = match get_cmake_build_type() {
+        Ok(CMakeBuildType::Debug) => (true, "Debug"),
+        Ok(CMakeBuildType::Release) => (false, "Release"),
+        Ok(CMakeBuildType::RelWithDebInfo) => (false, "RelWithDebInfo"),
+        Ok(CMakeBuildType::MinSizeRel) => (false, "MinSizeRel"),
         Err(e) => panic!("Cannot determine CMake build type: {}", e),
     };
 
@@ -99,32 +99,37 @@ fn main() {
         }
     }
 
-    let (out_dir, out_name) = if cfg!(all(windows, target_env = "msvc")) {
+    let mut out_dir = "./build".to_string();
+    if cfg!(all(windows, target_env = "msvc")) {
+        out_dir.push_str("/");
+        out_dir.push_str(win_folder);
+    }
+    let out_name = if cfg!(all(windows, target_env = "msvc")) {
         if is_debug {
             if cfg!(feature = "secure") {
-                ("./build/Debug", "mimalloc-static-secure-debug")
+                "mimalloc-static-secure-debug"
             } else {
-                ("./build/Debug", "mimalloc-static-debug")
+                "mimalloc-static-debug"
             }
         } else {
             if cfg!(feature = "secure") {
-                ("./build/Release", "mimalloc-static-secure")
+                "mimalloc-static-secure"
             } else {
-                ("./build/Release", "mimalloc-static")
+                "mimalloc-static"
             }
         }
     } else {
         if is_debug {
             if cfg!(feature = "secure") {
-                ("./build", "mimalloc-secure-debug")
+                "mimalloc-secure-debug"
             } else {
-                ("./build", "mimalloc-debug")
+                "mimalloc-debug"
             }
         } else {
             if cfg!(feature = "secure") {
-                ("./build", "mimalloc-secure")
+                "mimalloc-secure"
             } else {
-                ("./build", "mimalloc")
+                "mimalloc"
             }
         }
     };

